src/os/bluestore/BlueStore.cc

```diff
@@ -4447,15 +4447,19 @@ void BlueStore::MempoolThread::_update_cache_settings()
 #define dout_prefix *_dout << "bluestore.OmapIteratorImpl(" << this << ") "
 
 BlueStore::OmapIteratorImpl::OmapIteratorImpl(
-  PerfCounters* _logger, CollectionRef c, OnodeRef& o, KeyValueDB::Iterator it)
+  PerfCounters* _logger, CollectionRef c, OnodeRef& o, KeyValueDB::Iterator it, const omap_iter_seek_t& start_from)
   : logger(_logger), c(c), o(o), it(it)
 {
   logger->inc(l_bluestore_omap_iterator_count);
   std::shared_lock l(c->lock);
   if (o->onode.has_omap()) {
-    o->get_omap_key(string(), &head);
+    o->get_omap_key(start_from.key, &head);
     o->get_omap_tail(&tail);
-    it->lower_bound(head);
+    if (start_from.lower_bound) {
+      it->lower_bound(head);
```

Comment:
s/head/start_from.key/

Reply:
scratch that.
It should be first
`o->get_map_key(start_from.key, &a_key)`
and then....
`it->lower_bound(a_key)`

Reply:
Should be enough to drop https://github.com/ceph/ceph/pull/60000/commits/65b78befbe77e89e967a4b50e9f4e0436863fdf0 the bound hint's changes have been dissected into.